Resolve a few flaky Key Vault tests

heaths opened the pull request:
* Wait longer for deleted keys. Resolves a recent issue found during live tests.
* Fix net5.0 tests on OSX.
* Add EventSource for CertificateOperation. Adds more logging to better diagnose how #17718 is happening.

heaths committed:
Wait longer for deleted keys

heaths committed:
Fix net5.0 tests on OSX

heaths committed:
Add EventSource for CertificateOperation

Added more logging to better diagnose how #17718 is happening.

christothes commented on sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs:
```diff
@@ -261,5 +269,36 @@ public virtual async Task DeleteAsync(CancellationToken cancellationToken = defa
+                {
+                    EventSource.BeginUpdateStatus(_operation.Properties);
+
+                    _start = Stopwatch.GetTimestamp();
```
Do we need the StopWatch if we have the timestamps from `BeginUpdateStatus` and `EndUpdateStatus`?

heaths replied:
That's a good point. If a listener was interested, they could print out the timestamps or even calculate it themselves. I figured this was a convenient and relatively cheap way of getting elapsed times. @pakrym what do you think? Worth keeping? I know Azore.Core does this on 1 or 2 events.

Also, I realized last night a source could be enabled or disabled in the middle, so if I keep this I should always get the start time (relatively cheap) but can optimize out the double calculation if not enabled.

pakrym replied:
As far as I understand these events are added to diagnose possible test failures and will be removed afterward. I'm fine with whatever helps you diagnose the issue.

heaths replied:
Let's pretend/assume we keep them in. Does that change your mind? If I can't diagnose why these occasionally fail before GA, I might be inclined to leave them in.

I don't actually need the elapsed time so I'll just remove it.

pakrym replied:
Let's pretend/assume we keep them in. Does that change your mind? If I can't diagnose why these occasionally fail before GA, I might be inclined to leave them in.

If we keep them forever it's very inconsistent to have it only for one operation and not the others. I'm fine leaving it for GA but if the plan is to keep it long term we need a better cross-cutting design.

heaths replied:
I just removed it. It really wasn't necessary for timing anyway. Was just a thought. I agree a consistent solution is warranted. You have it for 1 task (general response, I believe?) but in that case it makes sense: easily determining how long a response took from a request seems like something very common - common enough not to make people diff timestamps themselves (especially with concurrent requests and having to find the associated request). Some guideline based on that logic might be a good start.

Here, it was just info. I did it because I could - not really any more reason than that. :)

heaths committed:
Always get start time from stopwatch

...in case the event source is initially disabled, but enabled in the middle (elapsed time would be very long).

heaths committed:
Fix broken test on OSX

heaths committed:
More fixes for OSX

heaths committed:
Resolve PR feedback

heaths committed:
Re-enable some tests and increasing polling interval

heaths committed:
Ignore VerifyECDecoderPrime256v1Imported on OSX
